pkg/pillar/nireconciler/linux_config.go

```diff
@@ -1652,7 +1675,7 @@ func (r *LinuxNIReconciler) getVLANConfigForNI(ni *niInfo) (
 }
 
 func (r *LinuxNIReconciler) getNISubnet(ni *niInfo) *net.IPNet {
-	if ni.config.Subnet.IP == nil {
+	if ni.config.Subnet == nil {
```

Comment:
Can Subnet.IP also be nil? Do you need to check for both?

Reply:
No, ParseCIDR either returns `nil` for the entire `net.IPNet` (and non-nil error) or both IP and Mask are set. So we (and other users of net.ParseCIDR) do not perform nil check for Subnet.IP.